fcrepo-http-commons/src/test/java/org/fcrepo/http/commons/api/rdf/HttpIdentifierConverterTest.java

```diff
@@ -0,0 +1,203 @@
+
+    @Test
+    public void testFirstLevelWithMemento() {
+        final String memento = "20190926133245";
+        final String baseUid = getUniqueId();
+        final String baseUrl = uriBase + "/" + baseUid;
+        final String testUri = baseUrl + "/" + FCR_VERSIONS + memento;
+        final String fedoraId = converter.convert(testUri);
+        assertEquals("info:fedora/" + baseUid, fedoraId);
+        final String httpUri = converter.reverse().convert(fedoraId);
+        assertEquals(baseUrl, httpUri);
+    }
+
+    @Test
+    public void testSecondLevel() {
+        final String baseUid = getUniqueId() + "/" + getUniqueId();
+        final String testUri = uriBase + "/" + baseUid;
+        final String fedoraId = converter.convert(testUri);
+        assertEquals("info:fedora/" + baseUid, fedoraId);
+        final String httpUri = converter.reverse().convert(fedoraId);
+        assertEquals(testUri, httpUri);
+    }
+
+    @Test
+    public void testSecondLevelWithAcl() {
+        final String baseUid = getUniqueId() + "/" + getUniqueId();
+        final String baseUrl = uriBase + "/" + baseUid;
+        final String testUri = baseUrl + "/" + FCR_ACL;
+        final String fedoraId = converter.convert(testUri);
+        assertEquals("info:fedora/" + baseUid, fedoraId);
+        final String httpUri = converter.reverse().convert(fedoraId);
+        assertEquals(baseUrl, httpUri);
+    }
+
+    @Test
+    public void testSecondLevelWithMetadata() {
+        final String baseUid = getUniqueId() + "/" + getUniqueId();
+        final String baseUrl = uriBase + "/" + baseUid;
+        final String testUri = baseUrl + "/" + FCR_METADATA;
+        final String fedoraId = converter.convert(testUri);
+        assertEquals("info:fedora/" + baseUid, fedoraId);
+        final String httpUri = converter.reverse().convert(fedoraId);
+        assertEquals(baseUrl, httpUri);
+    }
+
+    @Test
+    public void testSecondLevelWithVersions() {
+        final String baseUid = getUniqueId() + "/" + getUniqueId();
+        final String baseUrl = uriBase + "/" + baseUid;
+        final String testUri = baseUrl + "/" + FCR_VERSIONS;
+        final String fedoraId = converter.convert(testUri);
+        assertEquals("info:fedora/" + baseUid, fedoraId);
+        final String httpUri = converter.reverse().convert(fedoraId);
+        assertEquals(baseUrl, httpUri);
+    }
+
+    @Test
+    public void testSecondLevelWithMemento() {
+        final String memento = "20190926133245";
+        final String baseUid = getUniqueId() + "/" + getUniqueId();
+        final String baseUrl = uriBase + "/" + baseUid;
+        final String testUri = baseUrl + "/" + FCR_VERSIONS + memento;
+        final String fedoraId = converter.convert(testUri);
+        assertEquals("info:fedora/" + baseUid, fedoraId);
+        final String httpUri = converter.reverse().convert(fedoraId);
+        assertEquals(baseUrl, httpUri);
+    }
+
+    @Test
+    public void testItemWithDoubleAcl() {
```

Comment:
is this a thing?

Reply:
No but I wanted to remove that huge regex we currently use, so splitting seemed easiest. But I wanted to make sure it worked correctly. 😄